PJRT plugin cannot be compiled successfully

### What happened?

Seems that some APIs and these parameters in the IREE runtime has changed (e.g. `iree_hal_command_buffer_fill_buffer`, `IreeApi::hal_device_queue_execute` ...), but in PJRT plugin the old APIs still be used.

Maybe we can add the PJRT plugin to CI build/test workflow to prevent such breaking.

Note that currently I just tested the CUDA plugin, but same problems may exist in other plugins (cpu, rocm..).

```
  [0/2] Re-checking globbed directories...
  [1/5] Building CXX object src/iree_pjrt/cuda/CMakeFiles/iree_pjrt_cuda_client.objects.dir/client.cc.o
  FAILED: src/iree_pjrt/cuda/CMakeFiles/iree_pjrt_cuda_client.objects.dir/client.cc.o
  /usr/bin/clang++-17  -I/home/twice/iree -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core -I/home/twice/iree/integrations/pjrt/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/src -I/home/twice/iree/compiler/bindings/c/. -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/compiler/bindings/c/. -I/home/twice/iree/third_party/llvm-project/llvm/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/include -I/home/twice/iree/third_party/llvm-project/mlir/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/tools/mlir/include -I/home/twice/iree/third_party/llvm-project/lld/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/tools/lld/include -I/home/twice/iree/runtime/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src/iree/base/internal/flatcc -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src/iree/schemas -I/home/twice/iree/integrations/pjrt/third_party/pjrt_c_api -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/third_party/pjrt_c_api -I/usr/local/cuda/include -I/home/twice/iree/third_party/nccl -isystem /home/twice/iree/third_party/flatcc/include -O3 -DNDEBUG -std=gnu++17 -fPIC -fvisibility=hidden -pthread -I/home/twice/iree/third_party/flatcc/include/ -I/home/twice/iree/third_party/flatcc/include/flatcc/reflection/ -MD -MT src/iree_pjrt/cuda/CMakeFiles/iree_pjrt_cuda_client.objects.dir/client.cc.o -MF src/iree_pjrt/cuda/CMakeFiles/iree_pjrt_cuda_client.objects.dir/client.cc.o.d -o src/iree_pjrt/cuda/CMakeFiles/iree_pjrt_cuda_client.objects.dir/client.cc.o -c /home/twice/iree/integrations/pjrt/src/iree_pjrt/cuda/client.cc
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/cuda/client.cc:34:18: error: no member named 'allow_inline_execution' in 'iree_hal_cuda_device_params_t'
     34 |   default_params.allow_inline_execution = false;
        |   ~~~~~~~~~~~~~~ ^
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/cuda/client.cc:42:7: error: no matching function for call to 'iree_hal_cuda_driver_create'
     42 |       iree_hal_cuda_driver_create(driver_name, &default_params, &driver_options,
        |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:379:61: note: expanded from macro 'IREE_RETURN_IF_ERROR'
    379 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_IDENTITY_(__VA_ARGS__)))
        |                                                             ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:286:62: note: expanded from macro 'IREE_STATUS_IMPL_RETURN_IF_API_ERROR_'
    286 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_GET_EXPR_)(__VA_ARGS__))); \
        |                                                              ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:231:47: note: expanded from macro 'IREE_STATUS_IMPL_GET_EXPR_'
    231 | #define IREE_STATUS_IMPL_GET_EXPR_(expr, ...) expr
        |                                               ^~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/hal/drivers/cuda/api.h:120:31: note: candidate function not viable: no known conversion from 'iree_hal_cuda_device_params_t *' to 'const iree_hal_cuda_driver_options_t *' for 2nd argument
    120 | IREE_API_EXPORT iree_status_t iree_hal_cuda_driver_create(
        |                               ^
    121 |     iree_string_view_t identifier,
    122 |     const iree_hal_cuda_driver_options_t* options,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  2 errors generated.
  [2/5] Building CXX object src/iree_pjrt/common/CMakeFiles/iree_pjrt_common_common.objects.dir/api_impl.cc.o
  FAILED: src/iree_pjrt/common/CMakeFiles/iree_pjrt_common_common.objects.dir/api_impl.cc.o
  /usr/bin/clang++-17  -I/home/twice/iree -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core -I/home/twice/iree/integrations/pjrt/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/src -I/home/twice/iree/compiler/bindings/c/. -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/compiler/bindings/c/. -I/home/twice/iree/third_party/llvm-project/llvm/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/include -I/home/twice/iree/third_party/llvm-project/mlir/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/tools/mlir/include -I/home/twice/iree/third_party/llvm-project/lld/include -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/llvm-project/tools/lld/include -I/home/twice/iree/runtime/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src/iree/base/internal/flatcc -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/iree_core/runtime/src/iree/schemas -I/home/twice/iree/integrations/pjrt/third_party/pjrt_c_api -I/home/twice/iree/integrations/pjrt/python_packages/iree_cuda_plugin/build/cmake/third_party/pjrt_c_api -isystem /home/twice/iree/third_party/flatcc/include -O3 -DNDEBUG -std=gnu++17 -fPIC -fvisibility=hidden -pthread -I/home/twice/iree/third_party/flatcc/include/ -I/home/twice/iree/third_party/flatcc/include/flatcc/reflection/ -MD -MT src/iree_pjrt/common/CMakeFiles/iree_pjrt_common_common.objects.dir/api_impl.cc.o -MF src/iree_pjrt/common/CMakeFiles/iree_pjrt_common_common.objects.dir/api_impl.cc.o.d -o src/iree_pjrt/common/CMakeFiles/iree_pjrt_common_common.objects.dir/api_impl.cc.o -c /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/api_impl.cc
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/api_impl.cc:594:7: error: too many arguments to function call, expected 5, have 6
    589 |   IREE_RETURN_IF_ERROR(IreeApi::hal_device_queue_execute(
        |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    590 |       device_.device(), IREE_HAL_QUEUE_AFFINITY_ANY,
    591 |       /*wait_semaphore_list=*/iree_hal_fence_semaphore_list(ready_fence_.get()),
    592 |       /*signal_semaphore_list=*/
    593 |       iree_hal_fence_semaphore_list(dst_buffer_ready_fence.get()), transfer_cb,
    594 |       iree_hal_buffer_binding_table_empty()));
        |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:379:61: note: expanded from macro 'IREE_RETURN_IF_ERROR'
    379 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_IDENTITY_(__VA_ARGS__)))
        |                                                             ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:286:62: note: expanded from macro 'IREE_STATUS_IMPL_RETURN_IF_API_ERROR_'
    286 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_GET_EXPR_)(__VA_ARGS__))); \
        |                                                              ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:231:47: note: expanded from macro 'IREE_STATUS_IMPL_GET_EXPR_'
    231 | #define IREE_STATUS_IMPL_GET_EXPR_(expr, ...) expr
        |                                               ^~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/iree_helpers.h:138:15: note: 'hal_device_queue_execute' declared here
    138 | iree_status_t hal_device_queue_execute(
        |               ^
    139 |     iree_hal_device_t* device, iree_hal_queue_affinity_t queue_affinity,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    140 |     const iree_hal_semaphore_list_t wait_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    141 |     const iree_hal_semaphore_list_t signal_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    142 |     iree_hal_command_buffer_t* command_buffer) {
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/api_impl.cc:838:24: error: no matching function for call to 'iree_hal_command_buffer_fill_buffer'
    838 |   IREE_RETURN_IF_ERROR(iree_hal_command_buffer_fill_buffer(
        |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:379:61: note: expanded from macro 'IREE_RETURN_IF_ERROR'
    379 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_IDENTITY_(__VA_ARGS__)))
        |                                                             ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:286:62: note: expanded from macro 'IREE_STATUS_IMPL_RETURN_IF_API_ERROR_'
    286 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_GET_EXPR_)(__VA_ARGS__))); \
        |                                                              ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:231:47: note: expanded from macro 'IREE_STATUS_IMPL_GET_EXPR_'
    231 | #define IREE_STATUS_IMPL_GET_EXPR_(expr, ...) expr
        |                                               ^~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/hal/command_buffer.h:717:31: note: candidate function not viable: requires 5 arguments, but 7 were provided
    717 | IREE_API_EXPORT iree_status_t iree_hal_command_buffer_fill_buffer(
        |                               ^
    718 |     iree_hal_command_buffer_t* command_buffer, iree_hal_buffer_ref_t target_ref,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    719 |     const void* pattern, iree_host_size_t pattern_length,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    720 |     iree_hal_fill_flags_t flags);
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/api_impl.cc:851:7: error: too many arguments to function call, expected 5, have 6
    845 |   IREE_RETURN_IF_ERROR(IreeApi::hal_device_queue_execute(
        |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    846 |       device(), IREE_HAL_QUEUE_AFFINITY_ANY,
    847 |       /*wait_semaphore_list=*/
    848 |       {1, &transfer_timeline_, &signal_alloca_complete},
    849 |       /*signal_semaphore_list=*/
    850 |       {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb,
    851 |       iree_hal_buffer_binding_table_empty()));
        |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:379:61: note: expanded from macro 'IREE_RETURN_IF_ERROR'
    379 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_IDENTITY_(__VA_ARGS__)))
        |                                                             ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:286:62: note: expanded from macro 'IREE_STATUS_IMPL_RETURN_IF_API_ERROR_'
    286 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_GET_EXPR_)(__VA_ARGS__))); \
        |                                                              ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:231:47: note: expanded from macro 'IREE_STATUS_IMPL_GET_EXPR_'
    231 | #define IREE_STATUS_IMPL_GET_EXPR_(expr, ...) expr
        |                                               ^~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/iree_helpers.h:138:15: note: 'hal_device_queue_execute' declared here
    138 | iree_status_t hal_device_queue_execute(
        |               ^
    139 |     iree_hal_device_t* device, iree_hal_queue_affinity_t queue_affinity,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    140 |     const iree_hal_semaphore_list_t wait_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    141 |     const iree_hal_semaphore_list_t signal_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    142 |     iree_hal_command_buffer_t* command_buffer) {
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/api_impl.cc:1195:7: error: too many arguments to function call, expected 5, have 6
   1189 |   IREE_RETURN_IF_ERROR(IreeApi::hal_device_queue_execute(
        |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   1190 |       device(), IREE_HAL_QUEUE_AFFINITY_ANY,
   1191 |       /*wait_semaphore_list=*/
   1192 |       {1, &transfer_timeline_, &signal_alloca_complete},
   1193 |       /*signal_semaphore_list=*/
   1194 |       {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb,
   1195 |       iree_hal_buffer_binding_table_empty()));
        |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:379:61: note: expanded from macro 'IREE_RETURN_IF_ERROR'
    379 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_IDENTITY_(__VA_ARGS__)))
        |                                                             ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:286:62: note: expanded from macro 'IREE_STATUS_IMPL_RETURN_IF_API_ERROR_'
    286 |       IREE_STATUS_IMPL_IDENTITY_(IREE_STATUS_IMPL_GET_EXPR_)(__VA_ARGS__))); \
        |                                                              ^~~~~~~~~~~
  /home/twice/iree/runtime/src/iree/base/status.h:231:47: note: expanded from macro 'IREE_STATUS_IMPL_GET_EXPR_'
    231 | #define IREE_STATUS_IMPL_GET_EXPR_(expr, ...) expr
        |                                               ^~~~
  /home/twice/iree/runtime/src/iree/base/status.h:230:41: note: expanded from macro 'IREE_STATUS_IMPL_IDENTITY_'
    230 | #define IREE_STATUS_IMPL_IDENTITY_(...) __VA_ARGS__
        |                                         ^~~~~~~~~~~
  /home/twice/iree/integrations/pjrt/src/iree_pjrt/common/iree_helpers.h:138:15: note: 'hal_device_queue_execute' declared here
    138 | iree_status_t hal_device_queue_execute(
        |               ^
    139 |     iree_hal_device_t* device, iree_hal_queue_affinity_t queue_affinity,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    140 |     const iree_hal_semaphore_list_t wait_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    141 |     const iree_hal_semaphore_list_t signal_semaphore_list,
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    142 |     iree_hal_command_buffer_t* command_buffer) {
        |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  4 errors generated.
  ninja: build stopped: subcommand failed.
```

### Steps to reproduce your issue

Follow the instructions in PJRT plugin's README: https://github.com/iree-org/iree/blob/main/integrations/pjrt/README.md

- install jax and iree
- cd to `integrations/pjrt`
- `pip install -v --no-deps -e python_packages/iree_cuda_plugin`

### What component(s) does this issue relate to?

Other

### Version information

main branch (the newest commit).

### Additional context

_No response_

--- a/integrations/pjrt/src/iree_pjrt/common/api_impl.cc
+++ b/integrations/pjrt/src/iree_pjrt/common/api_impl.cc
@@ -590,8 +590,8 @@ iree_status_t BufferInstance::CopyToHost(void* dst, iree_host_size_t dst_size,
       device_.device(), IREE_HAL_QUEUE_AFFINITY_ANY,
       /*wait_semaphore_list=*/iree_hal_fence_semaphore_list(ready_fence_.get()),
       /*signal_semaphore_list=*/
-      iree_hal_fence_semaphore_list(dst_buffer_ready_fence.get()), transfer_cb,
-      iree_hal_buffer_binding_table_empty()));
+      iree_hal_fence_semaphore_list(dst_buffer_ready_fence.get()),
+      transfer_cb.get()));
 
   *out_done_event = copy_done_event;
   return iree_ok_status();
@@ -836,9 +836,8 @@ iree_status_t DeviceInstance::HostBufferToDeviceSplat(
       /*binding_capacity=*/0, &transfer_cb));
   IREE_CHECK_OK(iree_hal_command_buffer_begin(transfer_cb.get()));
   IREE_RETURN_IF_ERROR(iree_hal_command_buffer_fill_buffer(
-      transfer_cb.get(), buffer.get(), /*target_offset=*/0,
-      /*target_size=*/byte_length, data, element_type_byte_size,
-      IREE_HAL_FILL_FLAG_NONE));
+      transfer_cb.get(), iree_hal_make_buffer_ref(buffer.get(), 0, byte_length),
+      data, element_type_byte_size, IREE_HAL_FILL_FLAG_NONE));
   IREE_CHECK_OK(iree_hal_command_buffer_end(transfer_cb.get()));
 
   // Execute the enqueued splat:
@@ -847,8 +846,7 @@ iree_status_t DeviceInstance::HostBufferToDeviceSplat(
       /*wait_semaphore_list=*/
       {1, &transfer_timeline_, &signal_alloca_complete},
       /*signal_semaphore_list=*/
-      {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb,
-      iree_hal_buffer_binding_table_empty()));
+      {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb.get()));
 
   // Wrap in a buffer view and return:
   iree::vm::ref<iree_hal_buffer_view_t> result_buffer_view;
@@ -1191,8 +1189,7 @@ iree_status_t DeviceInstance::HostBufferToDevice(
       /*wait_semaphore_list=*/
       {1, &transfer_timeline_, &signal_alloca_complete},
       /*signal_semaphore_list=*/
-      {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb,
-      iree_hal_buffer_binding_table_empty()));
+      {1, &transfer_timeline_, &signal_copy_complete}, transfer_cb.get()));
 
   // Wrap in a buffer view and return.
   iree::vm::ref<iree_hal_buffer_view_t> result_buffer_view;

--- a/integrations/pjrt/src/iree_pjrt/cuda/client.cc
+++ b/integrations/pjrt/src/iree_pjrt/cuda/client.cc
@@ -31,7 +31,6 @@ iree_status_t CUDAClientInstance::CreateDriver(iree_hal_driver_t** out_driver) {
   iree_hal_cuda_device_params_t default_params;
   iree_hal_cuda_device_params_initialize(&default_params);
   default_params.command_buffer_mode = IREE_HAL_CUDA_COMMAND_BUFFER_MODE_STREAM;
-  default_params.allow_inline_execution = false;
 
   // Driver params.
   iree_hal_cuda_driver_options_t driver_options;
@@ -39,7 +38,7 @@ iree_status_t CUDAClientInstance::CreateDriver(iree_hal_driver_t** out_driver) {
   driver_options.default_device_index = 0;
 
   IREE_RETURN_IF_ERROR(
-      iree_hal_cuda_driver_create(driver_name, &default_params, &driver_options,
+      iree_hal_cuda_driver_create(driver_name, &driver_options, &default_params,
                                   host_allocator_, out_driver));
   logger().debug("CUDA driver created");
   return iree_ok_status();

